src/components/createCredentialDialog/createCredentialDialog.js

```diff
@@ -240,6 +241,7 @@ const CreateCredentialDialog = ({
           errorMessage={t('form-dialog.label', { context: ['username', 'error'] })}
         >
           <TextInput
+            ouiaId="username"
```

Comment:
required: Since `ouiaId` is the same as the name property modify `TextInput` to apply the `ouiaId` automatically

- https://github.com/quipucords/quipucords-ui/blob/main/src/components/form/textInput.js#L126

```
...
const updatedOuiaId = ouiaId || updatedName;

<PfTextInput
   id={updatedId}
   name={updatedName}
   ouiaId={updatedOuiaId}
```

OR you could even make a `data-test` attribute if you're looking for consistency against something else (I say stick with ouiaId)

```
// example code only end-result may require alteration
const updatedDataTest = props['data-test'] || updatedName;
```

Doing this should clean up most of the changes that are similar around TextInput.

Reply:
I'm not sure about this. Our generated ids are not globally unique, and as a result, `data-ouia-component-id` starts to be duplicated in snapshots. This is probably less of a problem in app itself, where we pay more attention to use globally unique `name`s.

Also, while `ouiaId` might be the same as `name` in many cases, I **want** to keep them decoupled. I intend to use mostly `data-ouia-component-id` in Camayoc, and I don't want to be forced to update Camayoc immediately every time names change in UI. I don't yet know if this is a viable approach, but I would like to give it a try.

Reply:
Given the implementation of the code, decoupled still works with the update where `ouiaId` is the generated "fallback" and overridden by whatever custom ouiaID you input. Its the best of both, and appears to also cover cases where overlooked fields also exist

Reply:
Added in 0bcb6d08763d75a83d68f73ce10cc10e079e597f , along with changes to TextArea, discussed below.

I don't like how we lose unique ouia ids in snapshots, but as long as CI is happy, this is not a hill to die on.

Reply:
I meant before this PR, `data-ouia-component-id` in snapshot was globally unique. Now you can notice some repetitions, when multiple components in the same snapshot have `generatedid-` value.

Obviously, I am not aware of some larger architecture choices that impact this result, and how it changed historically.

[ouia spec says ids must be unique withing surrounding context](https://ouia.readthedocs.io/en/latest/README.html#ouia-component), not globally unique (like standard HTML `id` attribute), and I guess our in snapshot still are. And even if not, it doesn't really impact anything.

All in all, this is just something that gets under my skin and what I personally dislike. Since you are OK with this change, and there is no actual technical problem with it, I think it's best to move forward with what we have in that regard.

Reply:
Ah, so "generatedid-" is for testing snapshots only... we removed the random hash that gets appended otherwise the snapshots would be updating on every test